converter/internal/otelcolconvert/converter_helpers.go

```diff
@@ -48,7 +48,14 @@ func toTokenizedConsumers(components []componentID) []otelcol.Consumer {
 // in an internal package.
 func encodeMapstruct(v any) map[string]any {
 	var res map[string]any
-	if err := mapstructure.Decode(v, &res); err != nil {
+	var decoderConfig mapstructure.DecoderConfig = mapstructure.DecoderConfig{Squash: true, Result: &res}
```

Comment:
I kept this in as we discussed in case there are squashed fields we need it for. Since we don't specifically need it yet I can remove this change if we prefer.